### PR TITLE
updated moes switch convertors

### PR DESCRIPTION
**Pull request by vladispavlov**
I have two moes switches with optional neutral
https://www.aliexpress.com/item/1005002631613176.html

The current convertor doesn't allow to change power on behavior and indicator light state, but this functionality is available in the tuya app. So I checked datapoints and wrote an updated converter. It works fine.

### CODE REVIEW

**Comment by Koenkk** [comment]
thanks!

File: lib/tuya.js [hunk 1]
```diff
@@ -731,6 +734,19 @@ const ZMAM02 = {
...
+    powerOnBehavior: {
+        0: 'off',
+        1: 'on',
+        2: 'previous',
+    },
+    indicateLight: {
+        0: 'off',
+        1: 'switch',
+        2: 'position',
+    },
```
Comment (javicalle) [inline lib/tuya.js:747]: Maybe not related, but I have a Moes switch that accept this posible values:
```
    off = 0x00
    light_when_on = 0x01
    light_when_off = 0x02
    freeze = 0x03
```
The freeze value keeps the current backlight state.

Comment (vladispavlov) [reply lib/tuya.js:747]: I checked it in the tuya app and there are 3 values as it was defined.
<details><summary>screenshot</summary></details>

But I tested the "freeze" value and it works. I'll create a new pr. Thanks!

**Comment by ciB89** [comment]
Should this device with neutral wire not also work as a repeater?

**Comment by vladyspavlov** [comment]
> Should this device with neutral wire not also work as a repeater?

It depends on firmware. It wouldn't work as repeater with neutral line on current firmware.

**Pull request merged**